packages/mongo-nav/src/mongo-select/ProjectSelect.tsx

```diff
@@ -139,64 +138,67 @@ function ProjectSelect({
     }
   };
 
-  const fetchProjectsAsAdmin = (searchTerm = '') => {
-    const queryString = searchTerm ? `?term=${searchTerm}` : '';
-    const endpointURI = `${hosts.cloud}/user/shared/projects/search${queryString}`;
-    return fetch(endpointURI, {
-      credentials: 'include',
-      mode: 'cors',
-      method: 'GET',
-    });
-  };
-
-  const filterDataAsAdmin = useCallback(
-    debounce((fetchValue: string) => {
-      fetchProjectsAsAdmin(fetchValue)
-        .then(response => response.json())
-        .then((response: Array<ProjectFilterResponse>) => {
-          const data = response.map(({ cid: projectId, gn: projectName }) => ({
-            projectId,
-            projectName,
-          }));
+  const hasOnChangeProp = !!onChangeProp;
 
-          setFilteredData(data);
-          setIsFetching(false);
-        })
-        .catch(console.error);
-    }, 300),
-    [],
-  );
-
-  const filterData = () => {
-    const normalizedValue = value.toLowerCase();
-    const filtered = data?.filter(
-      datum => datum.projectName.toLowerCase().indexOf(normalizedValue) !== -1,
-    );
+  useEffect(() => {
```

Comment:
Seems closer to being extractable to a custom hook that can be shared between the org and project select components. Do you think it's possible (and a good idea) to do so?

Reply:
Yeah, that was my initial goal and I thought hard about it, but I couldn't think of a way to do it all at once. Hoping this is an incremental step towards more shared code.